src/persistence/StorageNode.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Time } from "../time/Time";
+import { Storage } from "./Storage";
+import { readFile, writeFile } from "fs/promises";
+import { Logger } from "../log/Logger";
+import { ByteArray } from "@project-chip/matter.js";
+import {StorageInMemory} from "./StorageInMemory";
+
+/** We store changes 1s after a value was set to the storage, but not more often than every 1s. */
+const COMMIT_DELAY = 1000 /* 1s */;
+
+const logger = Logger.get("StorageNode");
+
+export class StorageNode extends StorageInMemory {
+    private readonly commitTimer = Time.getTimer(COMMIT_DELAY, () => this.commit());
+    private waitForCommit = false;
+    private allowToCommit = false;
+
+    constructor(
+        private readonly path: string,
+    ) {
+        super();
+    }
+
+    async initialize() {
+        if (Object.keys(this.store).length > 0) {
+            throw new Error("Storage contains already values, can not initialize!");
+        }
+        try {
```

Comment:
The code will be more readable if you make an explicit check if the file exists...
I guess this is what the try / catch is doing but it is not clear.

Reply:
The best practice way also by Node.js official documentation is to do it that way, because exists (even the Sync variant) is kind of async when it comes to program flow, so there can be edge cases.

see https://nodejs.org/api/fs.html#fsexistspath-callback

> Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile(), or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.

Reply:
OK.

Since this logic is non-obvious, better to move it to an util class so it can be done consistently everywhere and doesn't increase the complexity of the code here.

For instance called FileSystem.ts. On it, create a method read that returns undefined if the file doesn't exist and add you reply above as a comment in the code

Reply:
I would not do that now because we onkly have this one place where we read a file